pkg/webhooks/machine_webhook.go

```diff
@@ -746,6 +746,21 @@ func validateAWS(m *machinev1beta1.Machine, config *admissionConfig) (bool, []st
 		)
 	}
 
+	// placementGroupPartition must be between 1 and 7 if it's not 0 (default).
+	if providerSpec.PlacementGroupPartition != 0 {
+		if providerSpec.PlacementGroupPartition < 1 || providerSpec.PlacementGroupPartition > 7 {
```

Comment:
So, actually, the valid range here is `< 0 || > 7` right? I think we can trim thing and bring it into one line. Leaving the comments to explain why it is that way